build_tools/benchmarks/common/benchmark_definition.py

```diff
@@ -11,30 +11,44 @@
 """
 
 import json
-import re
 import subprocess
 
 from dataclasses import dataclass
+from enum import Enum
 from typing import Any, Dict, Sequence
 
 __all__ = [
-    "AndroidDeviceInfo", "BenchmarkInfo", "BenchmarkResults", "BenchmarkRun",
-    "execute_cmd_and_get_output", "execute_cmd"
+    "DeviceType", "DeviceInfo", "BenchmarkInfo", "BenchmarkResults",
+    "BenchmarkRun", "execute_cmd_and_get_output", "execute_cmd"
 ]
 
+
+@dataclass
+class DriverInfo:
+  """An object describing a driver.
+
+  It includes the following characteristics:
+  - pretty_name: the pretty name, e.g., 'IREE-DyLib'
+  - arch_type: the targeted architecture type, e.g., 'CPU'
+  """
+
+  pretty_name: str
+  arch_type: str
```

Comment:
Can we find a more descriptive identifier than `arch_type`?  `arch` suggests "CPU architecture" as in x86 vs ARM, or "GPU architecture as in Pascal vs Kepler...  Since this seems to be coarser granularity than this, CPU vs GPU, I wonder if `target_type` or `device_type` would be more descriptive.  `device_type` is already taken below, but I also have a comment about that one below. `device` would potentially be the right term here as it's how a GPU or a CPU is called in the context of heterogeneous compute APIs, IIUC (Lei would know better though).  Otherwise, `target` is a more generic term that we could use.